sydent/threepid/bind.py

```diff
@@ -125,6 +125,22 @@ def _notify(self, assoc, attempt):
         else:
             logger.info("Successfully notified on bind for %s" % (mxid,))
 
+            # Only remove sent tokens when they've been successfully sent.
+            if assoc.get("invites"):
```

Comment:
why do we need this condition?

Reply:
Because `invites` can be empty here (or even absent, according to Sydent's implem, which I now realise is not compliant with the spec for `/onbind`). Though in this case it doesn't prevent anything more than a useless DB call.

Reply:
I'd prefer a useless DB call than an extra code path.

Reply:
Fair enough.